Core/include/Acts/Seeding/SeedFinderOrthogonal.hpp

```diff
@@ -0,0 +1,79 @@
+// This file is part of the Acts project.
+//
+// Copyright (C) 2022 CERN for the benefit of the Acts project
+//
+// This Source Code Form is subject to the terms of the Mozilla Public
+// License, v. 2.0. If a copy of the MPL was not distributed with this
+// file, You can obtain one at http://mozilla.org/MPL/2.0/.
+
+#pragma once
+
+#include "Acts/Geometry/Extent.hpp"
+#include "Acts/Seeding/InternalSeed.hpp"
+#include "Acts/Seeding/InternalSpacePoint.hpp"
+#include "Acts/Seeding/SeedFinderOrthogonalConfig.hpp"
+#include "Acts/Seeding/SeedFinderUtils.hpp"
+
+#include <array>
+#include <list>
+#include <map>
+#include <memory>
+#include <set>
+#include <string>
+#include <utility>
+#include <vector>
+
+namespace Acts {
+
+template <typename external_spacepoint_t>
+class SeedFinderOrthogonal {
+  ///////////////////////////////////////////////////////////////////
+  // Public methods:
+  ///////////////////////////////////////////////////////////////////
+
+ public:
+  using seed_t = Seed<external_spacepoint_t>;
+  using internal_sp_t = InternalSpacePoint<external_spacepoint_t>;
+  using tree_t = KDTree<3, const internal_sp_t *, float, std::array, 4>;
+
+  /// The only constructor. Requires a config object.
+  /// @param config the configuration for the Seedfinder
+  SeedFinderOrthogonal(
+      Acts::SeedFinderOrthogonalConfig<external_spacepoint_t> config);
+  ~SeedFinderOrthogonal() = default;
+  /**    @name Disallow default instantiation, copy, assignment */
+  //@{
+  SeedFinderOrthogonal() = delete;
+  SeedFinderOrthogonal(const SeedFinderOrthogonal<external_spacepoint_t> &) =
+      delete;
+  SeedFinderOrthogonal<external_spacepoint_t> &operator=(
+      const SeedFinderOrthogonal<external_spacepoint_t> &) = delete;
+  //@}
+
+  template <typename input_container_t, typename output_container_t>
+  void createSeeds(const input_container_t &spacePoints,
+                   std::back_insert_iterator<output_container_t> out_it) const;
+
+  template <typename input_container_t>
+  std::vector<seed_t> createSeeds(const input_container_t &spacePoints) const;
+
+ private:
+  template <std::size_t NDims>
+  tree_t createTree(
+      const std::vector<const internal_sp_t *> &spacePoints) const;
+
+  template <typename OutputIt>
+  void helper(const internal_sp_t &middle,
+              const std::vector<const internal_sp_t *> &bottom,
+              const std::vector<const internal_sp_t *> &top, OutputIt it) const;
+
+  template <std::size_t NDims, typename output_it_t>
+  void processM(const tree_t &tree, output_it_t out_it,
+                const typename tree_t::pair_t &middle_p) const;
```

Comment:
would prefer a less generic naming and/or comments what this does

Reply:
Renamed to `processFromMiddleSP`. Also, comments added.